keychain/keychain_unix.go

```diff
@@ -0,0 +1,15 @@
+// +build !windows
+
+package keychain
+
+import "github.com/tmc/keyring"
+
+func set(provider string, password []byte) (err error) {
+	return keyring.Set(KeyChainName, provider, string(password))
+}
+
+func get(provider string) (pw []byte, err error) {
+	pwString, err := keyring.Get(KeyChainName, provider)
```

Comment:
The error isn't handled here. This makes the app attempt to authenticate even when now password was given or when Ctrl+C is pressed at the password prompt.

Reply:
Why does it need to be handled here? What you are describing is handled in keychain/keychain.go where it's actually exported.

Reply:
Lines 39 - 44 to be precise.

Reply:
In Go there are no exceptions. Every time you call a function which returns an error, you should check if the error is not nil. Even if you got a valid result back from the function, you could **also** get an error at the same time. This is perfectly valid in Go.

In this specific case, the app's flow continues after pressing Ctrl+C for example at the password prompt. This could lead to unexpected results and strange user experience.

Reply:
>What you are describing is handled in keychain/keychain.go where it's actually exported.

This is not enough - if function C calls function B and function B calls function A and function A returns an error, you should have an error handling "chain":

```go
func A() error {
	return errors.New("Oh no!")
}

func B() error {
	err := A()
	if err != nil {
		return err
	}
}

func C() {
	err := B()
	if err != nil {
		log.Fatalf("Something broke")
	}
}
```

Reply:
The bottom line is - every time you have something like this:

```go
err := whatever()
```

The following line should **nearly always** be:

```go
if err != nil {
    // Do something.
}
```